File: PYTHON/Smartphone.py
from PerangkatElektronik import PerangkatElektronik

class Smartphone(PerangkatElektronik):
    def __init__(self, merk='', nama='', tahun=0, jenis_koneksi='', baterai=None, ram=None, storage=None, cpu=None, display=None, sensor=None):
        super().__init__(merk, nama, tahun)  # Memanggil constructor dari PerangkatElektronik
        self.jenis_koneksi = jenis_koneksi
        self.baterai = baterai  # Komponen baterai
        self.ram = ram  # Komponen RAM
        self.storage = storage  # Komponen storage
        self.cpu = cpu  # Komponen CPU
        self.display = display  # Komponen Display
        self.sensor = sensor  # Komponen Sensor

    # Setter & Getter untuk atribut tambahan
    def set_jenis_koneksi(self, jenis_koneksi):
        self.jenis_koneksi = jenis_koneksi

    def get_jenis_koneksi(self):
        return self.jenis_koneksi

    def get_baterai(self):
        return self.baterai

    def get_ram(self):
        return self.ram

    def get_storage(self):
        return self.storage

    def get_cpu(self):
        return self.cpu

    def get_display(self):
        return self.display

    def get_sensor(self):
        return self.sensor

File: CPP/Smartphone.cpp
#pragma once
#include "Komponen.cpp"
#include "Baterai.cpp"
#include "RAM.cpp"
#include "Storage.cpp"
#include "CPU.cpp"
#include "Display.cpp"
#include "Sensor.cpp"
#include "PerangkatElektronik.cpp"

class Smartphone : public PerangkatElektronik {
private:
    string jenis_koneksi;
    Baterai baterai;
    RAM ram;
    Storage storage;
    CPU cpu;
    Display display;
    Sensor sensor;
    
public:
    // Constructor
    Smartphone(string merk, string nama, int tahun, string jenis_koneksi, Baterai b, RAM r, Storage s, CPU c, Display d, Sensor sn) {
        this->setMerk(merk);
        this->setNama(nama);
        this->setTahun(tahun);
        this->jenis_koneksi = jenis_koneksi;
        this->baterai = b;
        this->ram = r;
        this->storage = s;
        this->cpu = c;
        this->display = d;
        this->sensor = sn;
    }

    // Getter & Setter
    void setJenisKoneksi(string jenis_koneksi) {
        this->jenis_koneksi = jenis_koneksi;
    }

    string getJenisKoneksi() {
        return jenis_koneksi;
    }

    Baterai getBaterai() {
        return baterai;
    }

    RAM getRAM() {
        return ram;
    }

    Storage getStorage() {
        return storage;
    }

    CPU getCPU() {
        return cpu;
    }

    Display getDisplay() {
        return display;
    }

    Sensor getSensor() {
        return sensor;
    }

    // Destructor
    ~Smartphone() {
        // Optional: add any clean up code if needed
    }
};
